spec/requests/api/v2/access_request_spec.rb

```diff
@@ -24,6 +24,115 @@
+
+    context 'when unauthenticated' do
+      before do
+        list_access_requests_route
+      end
+
+      let(:payload) { { email: 'foo@bar' } }
+
+      it { should have_http_status(:unauthorized) }
+    end
+
+    context 'when unauthorized' do
+      let(:unauthorised_user) { create(:user) }
+      let(:payload) { { email: unauthorised_user.email } }
+      let(:unauthorised_user_route) do
+        get "/api/v2/access_requests/#{unauthorised_user.id}/list",
+            headers: { 'HTTP_AUTHORIZATION' => credentials }
+      end
+
+
+      it "should raise an error" do
+        expect { unauthorised_user_route }.to raise_error Pundit::NotAuthorizedError
+      end
+    end
+
+    context 'when authorised' do
+      let(:first_organisation) { create(:organisation, name: 'First Organisation') }
+      let(:second_organisation) { create(:organisation, name: 'Second Organisation') }
+      let(:first_user) {
+        create(:user,
+               first_name: 'First',
+               last_name: 'User',
+               email: 'first_user@test.com',
+               organisations: [first_organisation])
+      }
+      let(:second_user) {
+        create(:user,
+               first_name: 'Second',
+               last_name: 'User',
+               email: "second_user@test.com",
+               organisations: [second_organisation])
+      }
+      let!(:first_access_request) {
+        create(:access_request,
+               first_name: 'test',
+               last_name: 'user',
+               email_address: first_user.email,
+               requester_email: second_user.email,
+               requester_id: second_user.id,
+               organisation: first_organisation.name,
+               reason: 'Need additional support',
+               request_date_utc: '2019-05-05 00:10:47 UTC',
+               status: 'requested')
+      }
+      let!(:second_access_request) {
+        create(:access_request,
+               first_name: 'test',
+               last_name: 'user',
+               email_address: 'test@user.com',
+               requester_email: first_user.email,
+               requester_id: first_user.id,
+               organisation: second_organisation.name,
+               reason: 'Leaving current role',
+               request_date_utc: '2019-05-05 00:10:48 UTC',
+               status: 'requested')
+      }
+      before do
+        list_access_requests_route
+      end
+
+      it 'JSON has a data section with the correct attributes' do
+        json_response = JSON.parse response.body
+        expect(json_response).to eq(
+          [
+            {
+              "id" => first_access_request.id,
+              "email_address" => "first_user@test.com",
+              "first_name" => "test",
+              "last_name" => "user",
+              "requester_email" => "second_user@test.com",
+              "requester_id" => second_user.id,
+              "status" => "requested",
+              "organisation" => "First Organisation",
```

Comment:
For most of these values you can pull them from the parent variable I.e. `first_organisation.name`

Reply:
Done. Thanks

Reply:
There is one exception. When JSON parses a datetime into a string it seems to output it in a slightly different format. However, it is the correct time so i've manually made that attribute by inputting the output value in json. 

before 

`'2019-05-05 00:10:48 UTC'`
after
`'2019-05-05T00:10:47.000Z'`
🤷‍♂